profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java

```diff
@@ -34,76 +32,40 @@
  * @author Steve Swinsburg (steve.swinsburg@gmail.com)
  *
  */
+@Setter
 @Slf4j
 public class ProfilePrivacyLogicImpl implements ProfilePrivacyLogic {
 
-	private Cache cache;
-	private final String CACHE_NAME = "org.sakaiproject.profile2.cache.privacy";
+	private ProfileConnectionsLogic connectionsLogic;
+	private ProfileDao dao;
+	private SakaiProxy sakaiProxy;
 
-	/**
- 	 * {@inheritDoc}
- 	 */
 	@Override
 	public ProfilePrivacy getPrivacyRecordForUser(final String userId) {
-		return getPrivacyRecordForUser(userId, true);
-	}
-	
-	/**
- 	 * {@inheritDoc}
- 	 */
-	@Override
-	public ProfilePrivacy getPrivacyRecordForUser(final String userId, final boolean useCache) {
-		
-		if(userId == null){
-			throw new IllegalArgumentException("Null argument in ProfileLogic.getPrivacyRecordForUser"); 
-		}
 		
-		//will stay null if we can't get or create one
-		ProfilePrivacy privacy = null;
-		
-		//check cache
-		if(useCache) {
-			if(cache.containsKey(userId)){
-				log.debug("Fetching privacy record from cache for: " + userId);
-				privacy = (ProfilePrivacy)cache.get(userId);
-				if(privacy != null) {
-				  return(privacy);
-				}
-				// This means that the cache has expired. evict the key from the cache
-				log.debug("Privacy cache appears to have expired for " + userId);
-				this.cacheManager.evictFromCache(this.cache, userId);
-			}
+		if (StringUtils.isBlank(userId)) {
+			throw new IllegalArgumentException("Invalid userid " + userId);
```

Comment:
throw new IllegalArgumentException("Invalid userid {}, userId);

Reply:
i don't think you can do that in a throw, can you?

Reply:
You are right! :)